api/v1/authentication_types.go

```diff
@@ -58,6 +60,10 @@ type AuthenticationSpec struct {
 	// LDAP contains the configuration needed to setup LDAP authentication.
 	// +optional
 	LDAP *AuthenticationLDAP `json:"ldap,omitempty"`
+
+	// AuthenticationDeployment configures the Authentication Deployment.
+	// +optional
+	AuthenticationDeployment *AuthenticationDeployment `json:"authenticationDeployment,omitempty"`
```

Comment:
I would have expected this to be called dexDeployment. Otherwise, if we were to add more deployments in the future for this CR, how are we going to distinguish them?